### PR TITLE
GPII-3457: Update to work with newer versions of Testem.

### PR DESCRIPTION
**Pull request by the-t-in-rtf**
See [GPII-3457](https://issues.gpii.net/browse/GPII-3457) for details.

### CODE REVIEW

**Comment by the-t-in-rtf** [comment]
If anyone would like to try this out, there's now a dev release: 2.1.10-dev.20190404T122608Z.b51705e.GPII-3457

**Comment by the-t-in-rtf** [comment]
OK to test.

**Comment by the-t-in-rtf** [comment]
Ok to test.

**Comment by the-t-in-rtf** [comment]
ok to test.

**Comment by the-t-in-rtf** [comment]
Ok to test.

**Comment by the-t-in-rtf** [comment]
Ok to test.

**Comment by the-t-in-rtf** [comment]
The last tests timed out oddly.  Ok to test.

**Comment by the-t-in-rtf** [comment]
The build was aborted.  Ok to test.

**Comment by the-t-in-rtf** [comment]
I suspect this will time out or be aborted:

```
==> windows10: 01:05:38.917:  Test case listener has not responded after 5000ms - at sequence pos 2 of 8 sequence element {
==> windows10:     "listener": "fluid.identity",
==> windows10:     "event": "{testEnvironment}.events.onFixturesConstructed"
==> windows10: } of fixture chrome: Open and inspect a 'safe rollup'...
```

I will verify things working locally shortly.

**Comment by the-t-in-rtf** [comment]
Same hang in a local Vagrant VM, investigating now.

**Comment by the-t-in-rtf** [comment]
The tests that were hanging were the gpii-webdriver based tests of the "rollup" without Testem.  Looks like CI hit the same problem that @amb26 and I were discussing a week or so ago, i.e. if you have the latest chromedriver, you also have to have the latest Chrome.  I added a step to the VM provisioning to update Chrome, which hopefully will fix the tests.

File: src/js/testem-component.js [hunk 1]
```diff
@@ -463,7 +452,7 @@ fluid.defaults("gpii.testem.base", {
             }
         },
         cwd: "@expand:fluid.module.resolvePath({that}.options.cwd)",
-        user_data_dir: "@expand:gpii.testem.resolveFluidModulePathSafely({that}.options.testemDir)",
+        //user_data_dir: "@expand:gpii.testem.resolveFluidModulePathSafely({that}.options.testemDir)",
```
Comment (amb26) [inline src/js/testem-component.js]: Stray comments here, and on priority: first later on

Comment (the-t-in-rtf) [reply src/js/testem-component.js]: Thanks for catching that, fixed.

**Comment by amb26** [comment]
On my bare Windows machine, I get a test hang like the one reported above, only it is indefinite - 
````
11:56:58.968:  Express started...

DevTools listening on ws://127.0.0.1:61194/devtools/browser/16c2dd18-0c4e-4297-a81d-c470803fb697
11:57:03.846:  Test case listener has not responded after 5000ms - at sequence pos 2 of 8 sequence element {
    "listener": "fluid.identity",
    "event": "{testEnvironment}.events.onFixturesConstructed"
} of fixture chrome: Open and inspect a 'safe rollup'...
````
Shortly before the hang I see a window briefly open and close. I've updated to the latest Chromedriver and Chrome - any suggestions for what I can try now?

**Comment by the-t-in-rtf** [comment]
@amb26, I have seen similar stuff in other projects, it may be that there are issues with recent "headless" chrome versions.  Can you try skipping headless chrome and see if that avoids the hang?  You can just add "Headless Chrome" [to the other browsers that are skipped](https://github.com/GPII/gpii-testem/blob/a323cbdb4cbad1bc05fccc4c7d435d8fbf6cf2e8/tests/harness.js#L44)

**Comment by the-t-in-rtf** [comment]
The last CI failure was a failure in provisioning the VM, and the lack of npm caused the linting stage to fail as well.  Ok to test.

**Comment by the-t-in-rtf** [comment]
Provisioning keeps failing because the chocolatey servers are down for maintenance (here's [their status page](https://status.chocolatey.org)).  Will try again later.

**Comment by the-t-in-rtf** [comment]
Chocolatey is back.  Ok to test.

**Comment by the-t-in-rtf** [comment]
I think the last build was left over from the previous commit.  The build to test the test reordering is running now.

**Comment by the-t-in-rtf** [comment]
Ok to test.

**Comment by the-t-in-rtf** [comment]
Finally past the rollup doldrums thanks to the new version of gpii-webdriver.  Let's run a couple more passes to confirm that we don't have some other lingering instability.  Ok to test.

**Comment by the-t-in-rtf** [comment]
I have cut the 2.1.10 release and updated the forward-facing version in master.  There are still known issues with Windows and (non headless) Firefox, check out [GPII-4145](https://issues.gpii.net/browse/GPII-4145) for updates and workarounds.

**Pull request merged**